src/algopy_testing/op.py

```diff
@@ -938,69 +1000,315 @@ def _get_asset_holding(
         else:
             raise ValueError(f"Invalid asset holding field: {field}")
 
-    @staticmethod
     def asset_balance(
-        a: algopy.Account | algopy.UInt64 | int, b: algopy.Asset | algopy.UInt64 | int, /
+        self, a: algopy.Account | algopy.UInt64 | int, b: algopy.Asset | algopy.UInt64 | int, /
     ) -> tuple[algopy.UInt64, bool]:
         import algopy
 
-        balance, exists = AssetHoldingGet._get_asset_holding(a, b, "balance")
+        balance, exists = self._get_asset_holding(a, b, "balance")
         return algopy.UInt64(balance) if exists else algopy.UInt64(0), exists
 
-    @staticmethod
     def asset_frozen(
-        a: algopy.Account | algopy.UInt64 | int, b: algopy.Asset | algopy.UInt64 | int, /
+        self, a: algopy.Account | algopy.UInt64 | int, b: algopy.Asset | algopy.UInt64 | int, /
     ) -> tuple[bool, bool]:
-        frozen, exists = AssetHoldingGet._get_asset_holding(a, b, "frozen")
+        frozen, exists = self._get_asset_holding(a, b, "frozen")
         return bool(frozen), exists
 
 
+AssetHoldingGet = _AssetHoldingGet()
+
+
 class _AppParamsGet:
-    def __getattr__(self, name: str) -> Any:
-        raise NotImplementedError(
-            f"AppParamsGet.{name} is currently not available as a native "
-            "`algorand-python-testing` type. Use your own preferred testing "
-            "framework of choice to mock the behaviour."
-        )
+    def _get_app_param_from_ctx(
+        self, a: algopy.Application | algopy.UInt64 | int, param: str
+    ) -> tuple[Any, bool]:
+        import algopy
+
+        from algopy_testing import get_test_context
+
+        context = get_test_context()
+
+        if not context:
+            raise ValueError(
+                "Test context is not initialized! Use `with algopy_testing_context()` "
+                "to access the context manager."
+            )
+
+        active_txn = context.get_active_transaction()
+        if not active_txn:
+            raise ValueError("No active transaction found to reference application")
+
+        is_index = isinstance(a, (algopy.UInt64 | int)) and int(a) < 1001
+        try:
+            app_id = active_txn.apps(a).id if is_index else getattr(a, "id", a)
+        except IndexError as e:
+            raise ValueError(f"Invalid app index for apps in active transaction: {a}") from e
+        app_data = context.get_application(int(app_id))
+
+        if app_data is None:
+            return None, False
```

Comment:
Can't return `None` here, after looking at the algod implementation. I think we can safely just return `UInt64(0)` anytime an asset or application doesn't exist

Reply:
Resolved, feel free to reopen if you think something else is missing